### PR TITLE
Added Initialisation Function

### PR DESCRIPTION
**Pull request by raj2729**
## Description

> Initialisation Function is set up to get various information about the host system, such as if various programs are installed : node, npm python, etc , check for which platform it is running.

## Changes

- Added file ```initial.js``` at location ```app/electron/```
- Executed the function in ```app/electron/main.js```
- Dependencies Installed  :  ```shelljs``` , ```electron-store```

## Issue

> Closes issue :  #4

## Impacted Area

> The ```config.json``` file containing information during the initialisation of the electron app is stored at : 
  ```C:\Users\[USER_NAME]\AppData\Roaming\ergo```

## Steps to test

Steps needed to reproduce the scenario from this change to validate if the pull request solve this issue

- require the function from ```app/electron/initial.js``` into ```app/electron/main.js``` and then call the function inside ```main.js``` , run the app ```npm run dev```
- Check the ```config.json``` file stored at :  ```C:\Users\[USER_NAME]\AppData\Roaming\ergo``` 

## Screenshots

- Screenshot of the config.json file : 

> image

### CODE REVIEW

File: app/electron/initial.js [hunk 1]
```diff
@@ -0,0 +1,71 @@
+// To check inbuilt programs are installed or not
+// Include shelljs to run shell commands in node
+const sh = require("shelljs");
```
Comment (YJDoc2) [inline app/electron/initial.js]: use multi line comment format /**/

File: app/electron/initial.js [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+    newElectronStore.set("platform.Linux", false);
+  } else if (process.platform === "linux") {
+    newElectronStore.set("platform.Windows", false);
+    newElectronStore.set("platform.Mac", false);
+    newElectronStore.set("platform.Linux", true);
+  } else {
+    newElectronStore.set("platform.Linux", false);
+    newElectronStore.set("platform.Mac", false);
+    newElectronStore.set("platform.Windows", true);
+  }
```
Comment (YJDoc2) [inline app/electron/initial.js]: only set appropriate platform, rather than setting all three

File: app/electron/initial.js [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+    },
+    {
+      key: "npm",
+      value: "npm -v",
+    },
+    {
+      key: "jupyter-notebook",
+      value: "jupyter-notebook --version",
+    },
+  ];
```
Comment (YJDoc2) [inline app/electron/initial.js]: Extract in separate file

**Review by YJDoc2 — CHANGES_REQUESTED** [review]
Make the requested changes

**Review by YJDoc2 — APPROVED** [review]
Seems fine

**Review by namandangi — APPROVED** [review]

**Pull request merged**